coap/adapter.go

```diff
@@ -42,16 +44,21 @@ var _ Service = (*adapterService)(nil)
 // Observers is a map of maps,
 type adapterService struct {
 	auth      mainflux.ThingsServiceClient
-	conn      *broker.Conn
+	pubsub    nats.PubSub
 	observers map[string]observers
 	obsLock   sync.Mutex
 }
 
 // New instantiates the CoAP adapter implementation.
-func New(auth mainflux.ThingsServiceClient, nc *broker.Conn) Service {
+func New(auth mainflux.ThingsServiceClient, url string, logger log.Logger) Service {
```

Comment:
Why not `func New(auth mainflux.ThingsServiceClient, pubsub messaging.PubSub) Service {`?

Reply:
The Nats wrapper requires a logger as shown [here](https://github.com/mainflux/mainflux/blob/5ca8495f35691b7c97736cedc873168fb1f7688f/pkg/messaging/nats/pubsub.go#L53)

Reply:
The Nats wrapper requires a logger as shown [here](https://github.com/mainflux/mainflux/blob/5ca8495f35691b7c97736cedc873168fb1f7688f/pkg/messaging/nats/pubsub.go#L53)

Reply:
You can use the logger from the main as it's done here: https://github.com/mainflux/mainflux/blob/master/cmd/mqtt/main.go#L148

Reply:
Implemented